### PR TITLE
Ramazan ince node hw3

### PR DESCRIPTION
**Pull request by RamazanInce**

### CODE REVIEW

File: week3/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,85 @@
...
+const app = express();
+
+app.use(bodyParser.json());
+
+app.get('/todo/:id', (request, response) => {
+  fs.readFile('./todos.json', 'utf8', (error, todo) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+        response.status(404).send('');
```
Comment (eidosam) [inline week3/homework/src/index.js]: This line should be moved to outer the `if...else`, so you always respond to user.

    if (error) {
      if (error.code === 'ENOENT') {
        console.log('no data found');
      } else console.error(error);
      response.status(404).send('');
    } else {
      const newList = todo.split('\n');
      const id = request.params.id;
      response.json({ result: newList[id] });
    }

File: week3/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,85 @@
...
+  });
+});
+
+app.delete('/todo/:id', (request, response) => {
+  const id = request.params.id;
+  fs.readFile('./todos.json', 'utf8', (error, todosBinary) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+        response.status(404).send('');
```
Comment (eidosam) [inline week3/homework/src/index.js]: Same comment as above. (Moving `response.status(404).send('');` to outside the `if...else`

File: week3/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,85 @@
...
+      } else console.error(error);
+    } else {
+      const todos = JSON.parse(todosBinary);
+      todos.splice(id - 1, 1);
+
+      fs.writeFile('./todos.json', JSON.stringify(todos), error => {
```
Comment (eidosam) [inline week3/homework/src/index.js]: The `error` also here should be checked, so you send the response to user accordingly. Like what you did in the previous routes.

File: week3/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,85 @@
...
+      } else console.error(error);
+    } else {
+      const todos = todosBinary.trim() !== '' ? JSON.parse(todosBinary) : [];
+      todos.push({
+        text: request.body.todo,
+        done: false,
+      });
+      
+      fs.writeFile('./todos.json', JSON.stringify(todos), error => {
+        console.error(error);
```
Comment (eidosam) [inline week3/homework/src/index.js]: Same comment as above, to check the error and send the right status.

File: week3/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,85 @@
...
+        response.status(201).json({ result: request.body.todo });
+        
+      });
+    }
+    
+  });
+});
+app.delete('/todo', (request, response) => {
+  fs.readFile('./todos.json', 'utf8', (err, data) => {
+    if (err) throw err;
```
Comment (eidosam) [inline week3/homework/src/index.js]: This will shutdown the server. Error should be handled like above.

File: week3/homework/src/index.js [hunk 6]
```diff
@@ -1,3 +1,85 @@
...
+    fs.writeFile('./todos.json', '', err => {
+      if (err) throw err;
```
Comment (eidosam) [inline week3/homework/src/index.js]: Same as above.

**Comment by wouterkleijn** [comment]
Thanks a lot @eidosam for making the necessary changes :) @RamazanInce Can you make sure to implement them?

File: week3/homework/src/index.js (hunk 1 above)
Comment (RamazanInce) [reply week3/homework/src/index.js]: Ok I will do ASAP.

**Comment by RamazanInce** [comment]
I will do all the changes on Monday and push it again.
@eidosam thanks for your effort and feedback.

File: week3/homework/src/index.js (hunk 1 above)
Comment (RamazanInce) [reply week3/homework/src/index.js]: Thanks @eidosam I have corrected it.

File: week3/homework/src/index.js (hunk 2 above)
Comment (RamazanInce) [reply week3/homework/src/index.js]: Done

File: week3/homework/src/index.js (hunk 3 above)
Comment (RamazanInce) [reply week3/homework/src/index.js]: Done

File: week3/homework/src/index.js (hunk 4 above)
Comment (RamazanInce) [reply week3/homework/src/index.js]: I did not get it

File: week3/homework/src/index.js (hunk 5 above)
Comment (RamazanInce) [reply week3/homework/src/index.js]: I did not get it

File: week3/homework/src/index.js (hunk 6 above)
Comment (RamazanInce) [reply week3/homework/src/index.js]: I did not get it

Comment (RamazanInce) [reply week3/homework/src/index.js]: Hi @eidosam I did the necessary changes, please can you review the HW again for me.
Thanks

**Comment by RamazanInce** [comment]
> Thanks a lot @eidosam for making the necessary comments :) @RamazanInce Can you make sure to implement them?

Hi @wouterkleijn I did it ans pushed for review. Thanks for your follow up.

**Pull request closed without merging**